clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -406,6 +406,10 @@ public void close() {
         } finally {
             super.close();
         }
+        Node coordinator;
+        while ((coordinator = coordinator()) != null && client.pendingRequestCount(coordinator) > 0) {
+            client.poll(Long.MAX_VALUE);
```

Comment:
We may need to be a little careful about a wakeup here. It's common to use wakeup in order to shutdown a consumer thread and this causes a race condition if `close()` is reached prior to the wakeup being triggered. If we had a relatively short timeout, maybe we could just disable wakeups?